pkg/scheduler/core/generic_scheduler.go

```diff
@@ -52,17 +51,25 @@ const (
 	minFeasibleNodesPercentageToFind = 5
 )
 
-// ErrNoNodesAvailable is used to describe the error that no nodes available to schedule pods.
-var ErrNoNodesAvailable = fmt.Errorf("no nodes available to schedule pods")
-
 // ScheduleAlgorithm is an interface implemented by things that know how to schedule pods
 // onto machines.
 // TODO: Rename this type.
 type ScheduleAlgorithm interface {
-	Schedule(context.Context, framework.Framework, *framework.CycleState, *v1.Pod) (scheduleResult ScheduleResult, err error)
+	// Schedule(context.Context, framework.Framework, *framework.CycleState, *v1.Pod) (scheduleResult ScheduleResult, err error)
```

Comment:
Comparing to Schedule(), unrolling  Extenders() might be a better starting point as it involves fewer private variables.